src/app/components/elements/panels/EventGenderDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { EventBookingDTO, UserRole } from "../../../../IsaacApiTypes";
+import { Col, Table, UncontrolledTooltip } from "reactstrap";
+import { asPercentage } from "../../../services";
+
+export const countEventDetailsByRole = (role: UserRole | undefined, eventBookings: EventBookingDTO[]) => {
```

Comment:
Why is `role` allowed to be `undefined`? You never use it like that. You only use this function to check for students or teachers. So this can be simplified.

And another thing to look into in the future is the usage of `undefined` vs `null`. Please spend some time reading a couple of articles on when to use one or another, because even though they're similar, they are not the same, there are nuances. For instance, I think that you'd pretty much never want to set something to `undefined`, since the very act of setting a variable means that it's defined, so the right thing to do would be to set it to `null`. In this case, given that it's the first argument of a function, when you want to denote the lack of a role, you would have passed `null`. And the only time I'd check for `undefined` is to verify that a variable has not been set. I could be wrong here, since I'm not a JS expert, but this was my understanding.

Reply:
Essentially this was done on purpose so that you could use `countEventDetailsByRole` and if the `role` was not specified then it would do the calculation on all the bookings, not just `STUDENT` or `TEACHER` bookings.  Perhaps it would be better to refactor this and make it an optional parameter?

Reply:
Yes, I figured that was the reason you did it like that, but my point is that you don't really use the function the way you've described it. You only use it with role = `STUDENT` or `TEACHER`. So you've created something that is more complicated than it needs to be. In a way, this is premature optimisation, and it would be better to keep it simple, as you may not need it in the future. Look at that, I managed to fit Premature Optimisation, KISS and YAGNI in the same sentence 😎
So in this case, I would make the parameter mandatory, so `user` is always `UserRole` and never `undefined`.